miasm2/analysis/depgraph.py

```diff
@@ -605,15 +650,12 @@ class DependencyGraph_NoMemory(DependencyGraph):
     - Soundness: all results are corrects
     - Completeness: all possible solutions are founds
     """
+    def __init__(self, ira):
+        super(DependencyGraph_NoMemory, self).__init__(ira)
+        self.cb_follow = [follow_simp_expr, follow_nomem_cst]
 
-    def _resolve_depNode(self, depnode):
-        """Compute and return the dependencies involved by @depnode"""
-        # Get inital elements
-        result = super(DependencyGraph_NoMemory, self)._resolve_depNode(depnode)
-
-        # If @depnode depends on a memory element, give up
-        for node in result:
-            if isinstance(node.element, m2_expr.ExprMem):
-                return set()
-
-        return result
+class DependencyGraph_NoCall_NoMemory(DependencyGraph):
```

Comment:
To avoid defining `n!` class for each callbacks possibility, a metaclass could be a solution. That is to say, a class generator which take in input the callback to use, or with more abstraction, the element to handle (`follow_call`, `follow_mem`, ...).